## 2. Randomness

## Exercise 2.1

### Task

It is somewhat interesting that (5 * 4 * 3 * 2 * 1) perfectly divides (10 * 9 * 8 * 7 * 6) - there's no remainder.

If we only wanted exactly four heads as opposed to five, the equivalent calculation would be (10 * 9 * 8 * 7) / (4 * 3 * 2 * 1).

Does that evenly divide too? What is the formula in general?

Does it always come out as a positive whole number?

### Solution

In [2]:
(10 * 9 * 8 * 7) // (4 * 3 * 2 * 1)

210

As in the case of `(10 * 9 * 8 * 7 * 6)//(5 * 4 * 3 * 2 * 1)`, `(10 * 9 * 8 * 7)` also perfectly divides by `(4 * 3 * 2 * 1)`. There are 210 ways of getting 4 heads in 10 coin flips. 

The general formula is $\binom{n}{k} = \frac{n!}{k!(n-k)!}$. It is called a *Binomial coefficient* (a.k.a, the Choose function). (Wikibooks 2022)

The binomial coefficient is the number of distinct subsets of size *k* of a set of size *n*. (Millwood 2012) Or in simpler words, it is the number of ways of picking *k* things from *n* possibilities. (Wolfram MathWorld n.d.)

Number of possibilities for getting 4 heands in 10 tries using the formula above.

$\binom{10}{4} = \frac{10!}{4!(10-4)!} = \frac{10!}{4!(6)!} = \frac{10 x 9 x 8 x 7 x 6 x 5 x 4 x 3 x 2 x 1}{4 x 3 x 2 x 1 x (6 x 5 x 4 x 3 x 2 x 1)} = \frac{10 x 9 x 8 x 7}{4 x 3 x 2 x 1} = \frac{5040}{24} = 210$  


The outcome is always a positive integer.

0 $\leq$ k $\leq$ n and both k & n are always whole numbers as both the number of distinctive subsets (things) and the size of the set (posibilities) must be positive integers.

### References

GitHub (2023). *Writing mathematical expressions.* Online at https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions

Millwood B, (2012). *Binomial Coefficients.* Online at https://nrich.maths.org/7713

Wikibooks (2022). *LaTeX/Mathematics.* Online at https://en.wikibooks.org/wiki/LaTeX/Mathematics#Symbols

Wolfram MathWorld (n.d.). *Binomial Coefficient.* Online at https://mathworld.wolfram.com/BinomialCoefficient.html